### PR TITLE
Add tests to exercise CORS protocol

### PR DESCRIPTION
**Pull request by msavy**
- Add tests to exercise CORS protocol
- Fix a trivial bug detected during tests
- Relocate messages.properties and ensure resource availability to tests

### CODE REVIEW

**Comment by msavy** [comment]
I suggest we don't merge until plugin resource loading issue is fixed.

File: cors-policy/pom.xml [hunk 1]
```diff
@@ -24,9 +24,28 @@
       <artifactId>apiman-gateway-engine-policies</artifactId>
       <scope>provided</scope>
     </dependency>
+    
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.11</version>
```
Comment (msavy) [inline cors-policy/pom.xml]: Can I extract these deps and versions into the parent pom, or do we want to keep each plugin independent? I wasn't sure what our strategy was going to be on this.

Comment (EricWittmann) [inline cors-policy/pom.xml]: Yeah let's make it a proper parent and manage all our dependency versions there instead of in each plugin.

Comment (msavy) [inline cors-policy/pom.xml]: Yep, that seems the right way to do it to me as well. I've made a PR, and will update the CORS testing PR when that's merged.

Comment (EricWittmann) [inline cors-policy/pom.xml]: OK merged.

**Comment by msavy** [comment]
This has now been rebased to use the parent pom for deps and I've refactored to use the FailureFactory component in CORS and KC policies.

**Pull request merged**